[Linter] Added description for nonminimal bool linter rule (#893)
aptos-labs/developer-docs: apps/nextra/pages/en/build/smart-contracts/linter.mdx

- `(&s).f` can be simplified to `s.f`
- `(&mut s).f = 42;` can be simplified to `s.f = 42;`

### `nonminimal_bool`

> Check for boolean expressions that can be simplified when a boolean literal (either true or false) is part of a binary or unary boolean operator. Examples:

Reviewer: ```suggestion
Check for boolean expressions that can be simplified when a boolean literal (either `true` or `false`) is part of a binary or unary boolean operator. Examples:
```

Author: Done

- `x && true` is logically equivalent to `x`
- `x || true` is logically equivalent to `true`
- `x => false` is logically equivalent to `!x`
- `x <==> true` is logically equivalent to `x`
- `! true` is logically equivalent to `false`

> Does NOT consider side-effects/short-circuiting in recommended simplifcations. Example:

Reviewer: ```suggestion
Does NOT consider side-effects/short-circuiting in the recommended simplifications. Example:
```

Author: Done

> - `1/0 && true` is logically equivalent to `true`, but applying this simplification affects program dynamics

Reviewer: ```suggestion
- `1/0 && true` is logically equivalent to `true`, but applying this simplification affects program semantics.
```

Author: Done
